pkg/sql/create.go

```diff
@@ -947,6 +947,12 @@ func resolveFK(
 			}
 		}
 		if !found {
+			// Avoid unexpected index builds from ALTER TABLE ADD CONSTRAINT.
+			if mode == sqlbase.ConstraintValidity_Unvalidated {
+				return errors.Errorf(
+					"foreign key columns %s must be the prefix of an index", colNames(srcCols),
```

Comment:
Hm. I'm not sure I follow.. I assume a user wouldn't *expect* to see an error at all, if they didn't realize they need an index on the fK columns?